doc: stepper: update stepper documentation (#79630)
zephyrproject-rtos/zephyr: doc/hardware/peripherals/stepper.rst

           #address-cells = <1>;
           #size-cells = <0>;

           poscmp_enable; test_mode; lock_gconf; /* ADI TMC Global configuration flags */
           clock-frequency = <DT_FREQ_M(16)>; /* Internal/External Clock frequency */

           motor: motor@0 {
               status = "okay";
               reg=<0>;

               /* common stepper controller settings */
               invert-direction;
               micro-step-res = <256>;

               /* ADI TMC stallguard settings specific to TMC5041 */
               activate-stallguard2;
               stallguard-velocity-check-interval-ms=<100>;
               stallguard2-threshold=<9>;
               stallguard-threshold-velocity=<500000>;

               /* ADI TMC ramp generator as well as current settings */
               vstart = <10>;
               a1 = <20>;
               v1 = <30>;
               d1 = <40>;
               vmax = <50>;
               amax = <60>;
               dmax = <70>;
               tzerowait = <80>;
               vhigh = <90>;
               vcoolthrs = <100>;
               ihold = <1>;
               irun = <2>;
               iholddelay = <3>;
           };

>     };

Reviewer: I would move this example to the dts binding for the `tmc5041` and reference it from here, to have more bindings added to reference in the future. The documentation should be agnostic to any specific stepper controller :)

Reviewer: something like: 

```
See examples in:

- :dtcompatible:`adc,tmc5041`
```

Author: I followed how they had done in sensors. But this way it looks more neat :-)

Discord
=======

Zephyr has a `stepper discord`_ channel for stepper related discussions, which
is open to all.

.. _stepper-api-reference:

API Reference
*************

A common set of functions which should be implemented by all stepper drivers.

.. doxygengroup:: stepper_interface

Additional functions for Trinamic stepper drivers.

.. doxygengroup:: trinamic_stepper_interface

.. _stepper discord:

>    https://discord.com/channels/720317445772017664/1278263869982375946

Reviewer: This should be converted to an "agnostic" list of APIs, to be expanded in the future:
```
Stepper controller specific APIs
********************************

Trinamic
========

.. doxygengroup:: trinamic_stepper_interface
```

.. _stepper_api:

Steppers
########

The stepper driver API provides a set of functions for controlling and configuring stepper drivers.

Configure Stepper Driver
========================

- **Enable** the stepper driver using :c:func:`stepper_enable`.
- Configure **micro-stepping resolution** using :c:func:`stepper_set_micro_step_res`
  and :c:func:`stepper_get_micro_step_res`.
- Configure **actual position a.k.a step count** in microsteps using :c:func:`stepper_set_actual_position`
  and :c:func:`stepper_get_actual_position`.
- Set **max velocity** in micro-steps per second using :c:func:`stepper_set_max_velocity`

Control Stepper
===============

- **Move by** +/- micro-steps a.k.a. **relative movement** using :c:func:`stepper_move`.
- **Move to** a specific position a.k.a. **absolute movement** using :c:func:`stepper_set_target_position`.

> - Run incessantly with a **constant velocity** in a certain direction until

Reviewer 2: ```suggestion
- Run continuously at a **constant velocity** in a specific direction until
```

  a stop is detected using :c:func:`stepper_enable_constant_velocity_mode`.
- Check if the stepper is **moving** using :c:func:`stepper_is_moving`.

Device Tree
===========

In the context of stepper controllers  device tree provides the initial hardware
configuration for stepper drivers on a per device level. Each device must specify
a device tree binding in Zephyr, and ideally, a set of hardware configuration options
for things such as current settings, ramp parameters and furthermore. These can then
be used in a boards devicetree to configure a stepper driver to its initial state.

See examples in:

- :dtcompatible:`adi,tmc5041`

Discord
=======

Zephyr has a `stepper discord`_ channel for stepper related discussions, which

> is open to all.

Reviewer 2: not convinced about adding this section tbh. Not that it is necessarily a bad idea, just that I'm not sure it makes sense to potentially duplicate the same kind of info in each driver/subsystem doc page rather than just assume people will find out about Discord on other more generalistic pages

Author: There are some pages that already have the discord link, that's why I thought of putting it over here as well. I think it would be good to have a shortcut like this.
